bundle/edu.gemini.spdb.reports.collection/src/main/java/edu/gemini/dbTools/odbState/ObservationState.java

```diff
@@ -144,42 +143,38 @@ public TooType getTooType() {
     }
 
     /**
-     * Gets a formated string that indicates the night during which the
+     * Gets a formatted string that indicates the night during which the
      * observation took place.
      * @param time utc time of the last observation event message
      * @return formatted string indicating the night during which the
      * observation took place.
      */
     private static String _getNight(final long time) {
-        final Calendar cal = GregorianCalendar.getInstance();
-        cal.setTimeInMillis(time);
-
-        // The calendar is in the local time zone.  If the time is before
-        // noon, then the night started on the previous day.  If the time
-        // is after noon, then the night ends on the next day.
-        final int ampm = cal.get(Calendar.AM_PM);
-
-        final Calendar endCal;
-
-        final int startDay;
-        final int endDay;
-        if (ampm == Calendar.AM) {
-            endCal = (Calendar) cal.clone();
-            endDay = cal.get(Calendar.DAY_OF_MONTH);
-            cal.add(Calendar.DAY_OF_MONTH, -1);
-            startDay = cal.get(Calendar.DAY_OF_MONTH);
+        // Determine when, in the local time zone, the time started:
+        // If the time is before noon, then the night started on the previous day.
+        // If the time is after noon, then the night ends on the next day.
+        // Note: AM is 0, PM is 1 by documentation for ChronoField.
+        final LocalDateTime dt = LocalDateTime.ofInstant(Instant.ofEpochMilli(time), ZoneId.systemDefault());
+        final int ampm = dt.get(ChronoField.AMPM_OF_DAY);
+
+        final LocalDateTime start;
+        final LocalDateTime end;
+        if (ampm == 0) {
+            start = dt.minus(1, ChronoUnit.DAYS);
+            end   = dt;
         } else {
-            startDay = cal.get(Calendar.DAY_OF_MONTH);
-            cal.add(Calendar.DAY_OF_MONTH, 1);
-            endCal = cal;
-            endDay = cal.get(Calendar.DAY_OF_MONTH);
+            start = dt;
+            end   = dt.plus(1, ChronoUnit.DAYS);
         }
 
+        final int startDay = start.get(ChronoField.DAY_OF_MONTH);
+        final int endDay   = end.get(ChronoField.DAY_OF_MONTH);
+
         final StringBuilder buf = new StringBuilder();
         buf.append(startDay).append("/").append(endDay).append(" ");
 
-        final DateFormat f = new SimpleDateFormat("MMM yyyy");
-        buf.append(f.format(endCal.getTime()));
+        final DateTimeFormatter f = DateTimeFormatter.ofPattern("MMM yyyy").withZone(ZoneId.of("UTC"));
+        buf.append(f.format(end));
```

Comment:
The old code seemed odd to me. We got a time in UTC, and then shoved it into a `GregorianCalendar` that has the local time zone set, I guess it was to determine if the time in the local timezone was AM / PM.

We then print using the formatter.
Actually, this seems to work properly if we cut off the `DateTimeFormatter...withZone` just fine, so perhaps we should leave it off to ensure consistency.

The hours test I included passes either way. Seems strange, but I won't pretend to fully understand the dynamics of the new Java time API.

Reply:
I suppose it works without the time zone because the time zone information is incorporated in the `LocalDateTime`.  What I'm not sure of is what time zone is used for the formatting in this case.

Reply:
Investigating further, it seems like in the original case (as well as the new case whether I specify with `ZoneId.of("UTC")` or nothing) the time is being printed in UTC.